validator/rpc/standard_api.go

```diff
@@ -693,3 +696,43 @@ func validatePublicKey(pubkey []byte) error {
 	}
 	return nil
 }
+
+// SetVoluntaryExit TBD
+func (s *Server) SetVoluntaryExit(ctx context.Context, req *ethpbservice.SetVoluntaryExitRequest) (*ethpbservice.SetVoluntaryExitResponse, error) {
+	if s.validatorService == nil {
+		return nil, status.Error(codes.FailedPrecondition, "Validator service not ready")
+	}
+	if err := validatePublicKey(req.Pubkey); err != nil {
+		return nil, status.Error(codes.FailedPrecondition, err.Error())
+	}
+	if s.wallet == nil {
+		return nil, status.Error(codes.FailedPrecondition, "No wallet found")
+	}
+	km, err := s.validatorService.Keymanager()
+	if err != nil {
+		return nil, err
+	}
+	cfg := accounts.PerformExitCfg{
+		ValidatorClient: s.beaconNodeValidatorClient,
+		NodeClient:      s.beaconNodeClient,
+		Keymanager:      km,
+	}
+	sve, err := client.CreateSignedVoluntaryExit(ctx, cfg.ValidatorClient, cfg.NodeClient, cfg.Keymanager.Sign, req.Pubkey)
+	if err != nil {
+		return nil, status.Error(codes.FailedPrecondition, "Could not create voluntary exit")
+	}
+	vi, err := s.beaconNodeValidatorClient.ValidatorIndex(ctx, &eth.ValidatorIndexRequest{PublicKey: req.Pubkey})
+	if err != nil {
+		return nil, status.Error(codes.FailedPrecondition, "Invalid validator index")
+	}
+
+	return &ethpbservice.SetVoluntaryExitResponse{
+		Data: &ethpbservice.SetVoluntaryExitResponse_SignedVoluntaryExit{
+			Message: &ethpbservice.SetVoluntaryExitResponse_SignedVoluntaryExit_VoluntaryExit{
+				Epoch:          strconv.FormatUint(uint64(req.Epoch), 10),
```

Comment:
this should not be from the request it should be from sve( the response) same for the value below